### ISSUE TITLE
Helm Lint RBAC Colon Issue

### ISSUE THREAD

**Issue by tbickford** [helm/helm#8835]
It appears there was a bug introduced in Helm 3.3.0 with linting with certain RBAC resources (ClusterRole, Role, ClusterRoleBinding, and RoleBinding). The lint command is erroring when with RBAC roles and bindings with resources with a name containing a colon `:` which should be a valid name (ie `system:controller:deployment-controller` - standard out of the box Kuberentes ClusterRoleBinding).  These resources should not have the DNS naming constraints on them when linting.

It also looks like the `kpt` project had similar issues: https://github.com/GoogleContainerTools/kpt/issues/899

Output of `helm version`:

```
==> Linting .
[ERROR] templates/clusterrolebinding.yaml: object name does not conform to Kubernetes naming requirements: "some-namespace:system:some-app": invalid metadata name, must match regex ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])+$ and the length must not longer than 253
```

**Comment by mattfarina** [comment]
Thanks for bringing this up. I, and I imagine many others, were not aware that some names are now path based instead of DNS subdomain based.

**Comment by Chili-Man** [comment]
I'm running in to this issue as well when trying to lint the metrics-server helm chart: https://github.com/helm/charts/blob/master/stable/metrics-server/templates/aggregated-metrics-reader-cluster-role.yaml

**Comment by evoio** [comment]
Are there any plans to address this in the near future? We too are affected when attempting to work with pre-deployed resources (we're using EKS), metrics server and traefik among others. Since this seems to be the convention followed by Kubernetes we would also like to name all our RBAC resources using the same convention to maintain consistency.

**Comment by bacongobbler** [comment]
I haven't seen anyone from the community look to address this issue. Feel free to contribute a PR, @evoio.

**Comment by lindhe** [comment]
This was harder than it looked. There's almost no documentation available for good regex for DNS Label names or path segment names.

**Comment by gazal-k** [comment]
looks like this was introduced by: https://github.com/helm/helm/pull/8011. I wonder if just tweaking the regex to `^(([A-Za-z0-9][:-A-Za-z0-9_.]*)?[A-Za-z0-9])+$` would do?

**Comment by lindhe** [comment]
I believe we need to handle the three cases separately, depending on which resource type is being handled, unfortunately. Just making it more permissive would risk missing cases it should catch.

Also they seem to have different required max length, unless I'm mistaken.

**Comment by bacongobbler** [comment]
I had to close https://github.com/helm/helm/pull/9118 due to inactivity from the PR author. I provided feedback in that PR. If someone wants to revive that work and address those comments, please feel free.

**Comment by lindhe** [comment]
Stay open please.

**Comment by bacongobbler** [comment]
Can someone confirm whether https://github.com/helm/helm/pull/9416 fixed this issue? That was made available in Helm 3.5.4.
